### ISSUE TITLE
--add-colorbar on the right and left gives unexpected results

### ISSUE THREAD

**Issue by ScottLindstrom** [ssec/polar2grid#587]
(Putting the colorbar on the top or bottom looks a-okay to me)
If I do a --colorbar-align right or --colorbar-align left, the default colors go from maximum at the bottom to minimum at the top (although the labels will go from minimum at bottom to maximum at top)  This surprised me.  I can get the labels right, at least, if I set --colorbar-max to the minimum value (shown at the top of the colorbar) and --colorbar-min to the maximum value (shown at the bottom of the colorbar), but the default values don't match the values shown in the map;  that is to say, the minimum value plotted is plotted as the maximum value in the colorbar, and the maximum value is plotted as the minimum value in the colorbar.
I was doing this with a -p flash_extent_density field derived from gridded GLM data.  See this image:  https://www.ssec.wisc.edu/~scottl/data/GOES-16_GLM_flash_extent_density_20230124_180000_Houston.png

**Comment by djhoese** [comment]
Thanks for filing this. Ok so the colors are flipped when on the left and right, but the numbers are correct. Right?

**Comment by ScottLindstrom** [comment]
Yes. Numbers are correct (low at bottom to high at top), colors are flipped.

**Comment by djhoese** [comment]
@ScottLindstrom Do you know if this has ever worked with Polar2Grid or Geo2Grid?

**Comment by ScottLindstrom** [comment]
@djhoese I don't know the answer to that.

**Comment by djhoese** [comment]
So I made a test python script that can be run even with the old Polar2Grid 2.3 which I think was released before G2G 1.0 and I still get this result:

```python
from pydecorate import DecoratorAGG
from PIL import Image
from trollimage.colormap import Colormap

img = Image.new("RGBA", (500, 300))
dec = DecoratorAGG(img)
dec.align_left()
#dec.align_right()
dec.write_vertically()

cmap = Colormap(values=[0.0, 1.0], colors=[[0.0, 0.0, 0.0], [1.0, 1.0, 1.0]])
dec.add_scale(colormap=cmap, extend=True, line="blue")
img.show()
```

![image (1)](https://user-images.githubusercontent.com/1828519/214938062-575105b6-ee3c-440d-8ac8-adf11cdb02e2.png)

That is, the color mapping is 0 -> black, 1 -> white, but white appears on the bottom. I think this has always been like this. I agree with you it should be the opposite, but I've also noticed the number ticks are backwards from what I would expect for flipped values. That is, if you have a colormap from 280K to 180K, the bottom tick will be 280K and the top will be 180K.

**Comment by djhoese** [comment]
This should be fixed when this pydecorate PR is merged, released, and Polar2Grid/Geo2Grid is told to use this new version:

https://github.com/pytroll/pydecorate/pull/27

**Comment by djhoese** [comment]
Closing this as it was fixed upstream and will be included in the next G2G 1.2 beta tarball and then the next official release.
